BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -409,27 +428,43 @@ open class SBAUserDefaultsProfileItem: SBAProfileItemBase {
     }
 }
 
-open class SBAUserProfileItem: SBAProfileItemBase {
-    lazy open var user: SBAUser = {
-        return SBAUser.shared
-    }()
+open class SBAUserProfileItem: SBAKeychainProfileItem, SBANameDataSource {
     
     override open var value: Any? {
+        
         get {
-            // need to special-case accessing properties that are defined in protocol extensions
-            // because they're not available via KVC
-            switch SBAUserProtocolExtension(sourceKey) {
-            case SBAUserProtocolExtension.givenName:
```

Comment:
given name is the default value that maps to "name" for the case where the first name field is used to describe the full name and the last name field is unused. I'm getting around the loop of this thing points to the user and the user points back at this (which results in infinite loops of wacky madness and is why I'm not a big fan of singletons) by *only* having the user drop through to the shared profile manager.

`SBAUser -> ProfileManager -> ProfileItems`

These items access the given/family name directly. This still requires a few keys that have to be defined in the json and match the hardcoded values *but* it allows for a flat list without circular references.